cartography/intel/semgrep/findings.py

```diff
@@ -0,0 +1,231 @@
+        sca_vuln["closestSafeDependency"] = dep_fix
+        ref_urls = vuln["advisory"].get("references", {}).get("urls", [])
+        if ref_urls:
+            sca_vuln["ref_urls"] = ",".join(ref_urls)
+        sca_vuln["openedAt"] = vuln.get("openedAt", None)
+        for usage in vuln.get("usages", []):
+            usage_dict = {}
+            usage_dict["SCA_ID"] = unique_id
+            usage_dict["findingId"] = usage["findingId"]
+            usage_dict["path"] = usage["location"]["path"]
+            usage_dict["startLine"] = usage["location"]["startLine"]
+            usage_dict["startCol"] = usage["location"]["startCol"]
+            usage_dict["endLine"] = usage["location"]["endLine"]
+            usage_dict["endCol"] = usage["location"]["endCol"]
+            usage_dict["url"] = usage["location"]["url"]
+            usages.append(usage_dict)
+        vulns.append(sca_vuln)
+    return vulns, usages
+
+
+@timeit
+def load_semgrep_deployment(
+    neo4j_session: neo4j.Session, deployment: Dict[str, Any], update_tag: int,
+) -> None:
+    logger.info(f"Loading Semgrep deployment info {deployment} into the graph...")
+    load(
+        neo4j_session,
+        SemgrepDeploymentSchema(),
+        [deployment],
+        lastupdated=update_tag,
+    )
+
+
+@timeit
+def load_semgrep_sca_vulns(
+    neo4j_session: neo4j.Session,
+    vulns: List[Dict[str, Any]],
+    deployment_id: str,
+    update_tag: int,
+) -> None:
+    logger.info(f"Loading {len(vulns)} Semgrep SCA vulns info into the graph.")
+    load(
+        neo4j_session,
+        SemgrepSCAFindingSchema(),
+        vulns,
+        lastupdated=update_tag,
+        DEPLOYMENT_ID=deployment_id,
+    )
+
+
+@timeit
+def load_semgrep_sca_usages(
+    neo4j_session: neo4j.Session,
+    usages: List[Dict[str, Any]],
+    deployment_id: str,
+    update_tag: int,
+) -> None:
+    logger.info(f"Loading {len(usages)} Semgrep SCA usages info into the graph.")
+    load(
+        neo4j_session,
+        SemgrepSCALocationSchema(),
+        usages,
+        lastupdated=update_tag,
+        DEPLOYMENT_ID=deployment_id,
+    )
+
+
+@timeit
+def cleanup(
+    neo4j_session: neo4j.Session, common_job_parameters: Dict[str, Any],
+) -> None:
+    logger.info("Running Semgrep SCA findings cleanup job.")
+    findings_cleanup_job = GraphJob.from_node_schema(
+        SemgrepSCAFindingSchema(), common_job_parameters,
+    )
+    findings_cleanup_job.run(neo4j_session)
+    logger.info("Running Semgrep SCA Locations cleanup job.")
+    locations_cleanup_job = GraphJob.from_node_schema(
+        SemgrepSCALocationSchema(), common_job_parameters,
+    )
+    locations_cleanup_job.run(neo4j_session)
+
+
+@timeit
+def sync(
+    neo4j_sesion: neo4j.Session,
+    semgrep_app_token: str,
+    update_tag: int,
+    common_job_parameters: Dict[str, Any],
+) -> None:
+    logger.info("Running Semgrep SCA findings sync job.")
+    semgrep_deployment = get_deployment(semgrep_app_token)
```

Comment:
Nit and non-blocker: consider splitting to sync_sca_vulns and sync_sca_usages to make it cleaner.